spatialprofilingtoolbox/db/stratification_puller.py

```diff
@@ -1,67 +1,79 @@
 """Retrieve outcome data for all studies."""
+from typing import cast
+
 import pandas as pd
+from psycopg2.extensions import cursor as Psycopg2Cursor
 
-from spatialprofilingtoolbox.db.database_connection import DatabaseConnectionMaker
 from spatialprofilingtoolbox.standalone_utilities.log_formats import colorized_logger
 
 logger = colorized_logger(__name__)
 
 
-class StratificationPuller(DatabaseConnectionMaker):
+class StratificationPuller:
     """Retrieve sample cohort data for all studies."""
-    def __init__(self, database_config_file):
-        super().__init__(database_config_file=database_config_file)
+
+    cursor: Psycopg2Cursor
+    stratification: dict | None
+
+    def __init__(self, cursor: Psycopg2Cursor):
+        self.cursor = cursor
         self.stratification = None
 
-    def pull(self):
-        self.stratification = self.retrieve_stratification()
+    def pull(self) -> None:
+        self.stratification = self._retrieve_stratification()
 
-    def get_stratification(self):
-        return self.stratification
+    def get_stratification(self) -> dict:
```

Comment:
Can we provide a more detailed type hint for the dict? Looks like it's a `dict[str, dict[str, DataFrame]]`.

Reply:
I think this is a good idea after refactoring so that the return type is actually controlled. But that is not really part of this PR. We could make a new issue if it is important.

Reply:
Nevermind, my comment applies to the other pullers not this one. You are right, this one has all final values dataframes.

Reply:
(and the nesting levels are all consistent)